### PR TITLE
Add websocket broadcast

### PR DESCRIPTION
**Pull request by DenisLutski**

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -34,4 +34,10 @@ config :logger, :console,
   format: "$date $time $metadata[$level] $message\n",
   metadata: [:sdk, :method, :url]
 
+config :ext, :ext,
+  ws: [
+    subscriptions: [:subscription_1, :subscription_1],
+    endpoint: "ExtWeb.Endpoint can be in atom"
```
Comment (atanych) [inline config/config.exs]: `can be atom` ?

File: lib/ext/ws/broadcast.ex [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  Let's see an example:
+    Ext.Ws.Broadcast.call([1, 2], :fake, %{message: "fake_message"})
+    Ext.Ws.Broadcast.call([%{id: 1}, %{id: 2}], :fake, %{message: "fake_message"})
+    Ext.Ws.Broadcast.call(1, :fake, %{message: "fake_message"})
+    Ext.Ws.Broadcast.call(:stable).(1, :fake, %{message: "fake_message"})
+  """
+
+  use Ext.Ws.Queue
+
+  @list_subscriptions Ext.Config.get([:ext, :ws, :subscriptions]) || []
```
Comment (atanych) [inline lib/ext/ws/broadcast.ex]: ого так можно ? 
это же ахуенно, так можно оперировать константами

Comment (atanych) [reply lib/ext/ws/broadcast.ex]: Let's also name as `@subscriptions` or `@subscription_list`

File: lib/ext/ws/event.ex [hunk 3]
```diff
@@ -0,0 +1,8 @@
+defmodule Ext.Ws.Event do
+  @derive {Jason.Encoder, except: [:__meta__, :__struct__]}
+
+  defstruct [:uuid, :data]
+
+  def init(%{uuid: uuid, data: data}), do: %__MODULE__{uuid: uuid, data: data}
+  def init(data), do: %__MODULE__{uuid: Ecto.UUID.generate(), data: data}
```
Comment (atanych) [inline lib/ext/ws/event.ex:7]: 🔥

File: lib/ext/ws/broadcast.ex [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    do: Enum.each(user_ids, &call(&1, subscription, data, stable))
+
+  def call(%{id: id}, subscription, event, stable) when subscription in @list_subscriptions,
+    do: call(id, subscription, event, stable)
+
+  def call(user_id, subscription, %Ext.Ws.Event{} = event, _stable) when subscription in @list_subscriptions,
+    do: {:ok, Absinthe.Subscription.publish(@endpoint, event, [{subscription, "private:#{user_id}"}])}
+
+  def call(user_id, subscription, data, stable) when subscription in @list_subscriptions do
+    event = data |> build_event() |> save_event(user_id, subscription, stable)
```
Comment (atanych) [inline lib/ext/ws/broadcast.ex]: build_event is redundant

File: lib/ext/ws/event.ex [hunk 5]
```diff
@@ -0,0 +1,8 @@
+defmodule Ext.Ws.Event do
+  @derive {Jason.Encoder, except: [:__meta__, :__struct__]}
+
+  defstruct [:uuid, :data]
+
+  def init(%{uuid: uuid, data: data}), do: %__MODULE__{uuid: uuid, data: data}
```
Comment (atanych) [inline lib/ext/ws/event.ex:6]: looks like this is enough 

```
def init(%__MODULE__{} = struct), do: struct
```

File: lib/ext/ws/broadcast.ex [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+  Broadcast live event to connected user by topic `private:<user_id>`
+
+  Let's see an example:
+    Ext.Ws.Broadcast.call([1, 2], :fake, %{message: "fake_message"})
+    Ext.Ws.Broadcast.call([%{id: 1}, %{id: 2}], :fake, %{message: "fake_message"})
+    Ext.Ws.Broadcast.call(1, :fake, %{message: "fake_message"})
+    Ext.Ws.Broadcast.call(:stable).(1, :fake, %{message: "fake_message"})
+  """
+
+  use Ext.Ws.Queue
```
Comment (atanych) [inline lib/ext/ws/broadcast.ex]: Do we really need that?

File: lib/ext/ws/event.ex (hunk 5 above)
Comment (DenisLutski) [reply lib/ext/ws/event.ex:6]: from redis we get map, not struct

File: config/config.exs (hunk 1 above)
Comment (DenisLutski) [reply config/config.exs]: yes

Comment (atanych) [reply config/config.exs]: I mean "ExtWeb.Endpoint can be in atom" -> "ExtWeb.Endpoint can be atom" :)


But if we use only Atom, let's add description: `ExtWeb.Endpoint should be atom`

File: lib/ext/ws/event.ex (hunk 5 above)
Comment (atanych) [reply lib/ext/ws/event.ex:6]: why we can not add to `first` sctruct?

Comment (atanych) [reply lib/ext/ws/event.ex:6]: actually, I see two cases:
1. We use first
2. we pass data as is and should add uuid to this map

**Review by atanych — APPROVED** [review]

**Pull request merged**